src/code-templates/libraries/logger/pino.logger.ts

```diff
@@ -20,23 +20,38 @@ export default class PinoLogger implements Logger {
             },
           }
         : undefined,
-    };
-    this.logger = pino(opts);
+    });
   }
 
-  debug(message: string, ...args: unknown[]): void {
-    this.logger.debug(message, ...args);
+  debug(message: string, metadata?: Record<any, unknown>): void {
+    if (metadata) {
+      this.#logger.debug(metadata, message);
+    } else {
+      this.#logger.debug(message);
+    }
   }
 
-  error(message: string, ...args: unknown[]): void {
-    this.logger.error(message, ...args);
+  error(message: string, metadata?: Record<any, unknown>): void {
+    if (metadata) {
+      this.#logger.error(metadata, message);
+    } else {
+      this.#logger.error(message);
+    }
   }
 
-  info(message: string, ...args: unknown[]): void {
-    this.logger.info(message, ...args);
+  info(message: string, metadata?: Record<any, unknown>): void {
+    if (metadata) {
+      this.#logger.info(metadata, message);
+    } else {
+      this.#logger.info(message);
+    }
   }
 
-  warning(message: string, ...args: unknown[]): void {
-    this.logger.warn(message, ...args);
+  warning(message: string, metadata?: Record<any, unknown>): void {
+    if (metadata) {
+      this.#logger.warn(metadata, message);
+    } else {
+      this.#logger.warn(message);
+    }
   }
```

Comment:
Not just always doing
```
this.#logger.<method>(metadata, message);
```
as it won't print the message when the metadata is undefined...

----

Tiny dive into Pino code 😄 

This is happening cause Pino expects the first parameter to be either an object or a string:
And when passing `undefined` it passes to the `format` function `undefined`:
https://github.com/pinojs/pino/blob/dc4e8546c0a4bedb087cbcccd9a48b56100b0ad7/lib/tools.js#L56

and the format function just returns the first parameter if the type is not a string
https://github.com/pinojs/quick-format-unescaped/blob/20ebf64c2f2e182f97923a423d468757b9a24a63/index.js#L21-L23

Reply:
@rluvaton You're not leaving blind spots, gotta love this